### PR TITLE
EZP-24765: Give possibility to set additional paths for a given theme

### PR DESCRIPTION
**Pull request by lolautruche**
> https://jira.ez.no/browse/EZP-27465

This patch makes it possible to add arbitrary Twig template directories to a theme from configuration.

This can be useful when you want to define templates from third-party bundles as part of one of your themes, or when upgrading your application in order to use eZ Platform design engine, and that your existing templates are not yet following the convention.

Practical use case would be to define a `standard` theme for templates provided by the kernel.

```yaml
ezdesign:
    design_list:
        my_design: [my_theme, some_other_theme]
    template_theme_paths:
        # FOSUserBundle templates will be part of "my_theme" theme
        my_theme:
            - '%kernel.root_dir%/../vendor/friendsofsymfony/user-bundle/Resources/views'
```

### CODE REVIEW

**Comment by lolautruche** [comment]
Ping @andrerom @bdunogier

File: doc/templates.md [hunk 1]
```diff
@@ -41,6 +41,25 @@ Default fallback order is the following:
...
+are not yet following the convention.
+
+```yaml
+ezdesign:
+    design_list:
+        my_design: [my_theme, some_other_theme]
+    template_theme_paths:
+        # FOSUserBundle templates will be part of "my_theme" theme
+        my_theme:
+            - '%kernel.root_dir%/../vendor/friendsofsymfony/user-bundle/Resources/views'
```
Comment (andrerom) [inline doc/templates.md]: On sf 3.3 there is also `kernel.project_dir`, but doesn't make much of a differance here.

Comment (lolautruche) [inline doc/templates.md]: I was tempted to use `%kernel.project_dir%` but as we don't require Symfony 3.3 yet, I preferred using the old fashion way 😛

File: bundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -29,6 +29,15 @@ public function getConfigTreeBuilder()
                         ->prototype('scalar')->end()
                     ->end()
                 ->end()
+                ->arrayNode('template_theme_paths')
+                    ->useAttributeAsKey('theme')
+                    ->example(['some_theme' => ['/var/foo/some_theme_dir', '/another/theme/dir']])
+                    ->info('Collection of template paths by theme.')
```
Comment (bdunogier) [inline bundle/DependencyInjection/Configuration.php]: Actually, I'd make templates plural, and update the setting name. I know we've spent time on it, but it sounds wrong now :)

Funnily enough, it is reversed in the container parameter's name: `ezdesign.templates_path_map` 😉

File: doc/templates.md [hunk 3]
```diff
@@ -41,6 +41,25 @@ Default fallback order is the following:
 
 > Bundle fallback order is the instantiation order in `AppKernel`.
 
+### Additional theme paths
+In addition to the convention described above, it is also possible to add arbitrary Twig template directories to a theme
```
Comment (bdunogier) [inline doc/templates.md]: template**s** ?

**Comment by lolautruche** [comment]
@bdunogier Fixed

**Comment by lolautruche** [comment]
@bdunogier ping ? 😃

**Pull request merged**